sklearn/base.py

```diff
@@ -239,10 +238,10 @@ def set_params(self, **params):
         for key, value in params.items():
             key, delim, sub_key = key.partition("__")
             if key not in valid_params:
+                local_valid_params = sorted(self.get_params(deep=False).keys())
```

Comment:
Why is a second call to `get_params` needed here,  could `valid_params` be used directly to define `local_valid_params`?

Reply:
`valid_params` include the nested param names which we do not want to display in this context.